Store fewer raw pointers in containers in Source/WebKitLegacy

cdumez opened the pull request:
#### c095d300811b451846fb81f29255b82d4f76b938
<pre>
Store fewer raw pointers in containers in Source/WebKitLegacy
<a href="https://bugs.webkit.org/show_bug.cgi?id=262188">https://bugs.webkit.org/show_bug.cgi?id=262188</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/history/HistoryItem.h:
* Source/WebKitLegacy/Storage/StorageNamespaceImpl.cpp:
(WebKit::localStorageNamespaceMap):
(): Deleted.
* Source/WebKitLegacy/Storage/StorageNamespaceImpl.h:
* Source/WebKitLegacy/Storage/StorageThread.cpp:
(WebCore::activeStorageThreads):
(WebCore::StorageThread::start):
(WebCore::StorageThread::terminate):
(): Deleted.
* Source/WebKitLegacy/Storage/StorageThread.h:
* Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.cpp:
(WebKit::storageNamespaceProviders):
(WebKit::WebStorageNamespaceProvider::WebStorageNamespaceProvider):
(WebKit::WebStorageNamespaceProvider::~WebStorageNamespaceProvider):
(WebKit::WebStorageNamespaceProvider::closeLocalStorage):
(WebKit::WebStorageNamespaceProvider::clearLocalStorageForAllOrigins):
(WebKit::WebStorageNamespaceProvider::clearLocalStorageForOrigin):
(WebKit::WebStorageNamespaceProvider::closeIdleLocalStorageDatabases):
(WebKit::WebStorageNamespaceProvider::syncLocalStorage):
(): Deleted.
* Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.h:
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.cpp:
(WebResourceLoadScheduler::hostForURL):
(WebResourceLoadScheduler::scheduleLoad):
(WebResourceLoadScheduler::remove):
(WebResourceLoadScheduler::crossOriginRedirectReceived):
(WebResourceLoadScheduler::servePendingRequests):
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.h:
* Source/WebKitLegacy/mac/History/BackForwardList.h:
* Source/WebKitLegacy/mac/History/WebBackForwardList.mm:
(kit):
(-[WebBackForwardList initWithBackForwardList:]):
(-[WebBackForwardList dealloc]):
* Source/WebKitLegacy/mac/History/WebHistoryItem.mm:
(-[WebHistoryItem dealloc]):
(-[WebHistoryItem copyWithZone:]):
(core):
(kit):
(-[WebHistoryItem initWithWebCoreHistoryItem:]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebVisitedLinkStore.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebVisitedLinkStore.mm:
(visitedLinkStores):
(WebVisitedLinkStore::WebVisitedLinkStore):
(WebVisitedLinkStore::~WebVisitedLinkStore):
(WebVisitedLinkStore::removeAllVisitedLinks):
* Source/WebKitLegacy/mac/WebView/WebScriptWorld.mm:
(-[WebScriptWorld initWithWorld:]):
(-[WebScriptWorld dealloc]):
(+[WebScriptWorld findOrCreateWorld:]):

Canonical link: <a href="https://commits.webkit.org/268556@main">https://commits.webkit.org/268556@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/84e3efa0236d2baf5527d8c0bb6020ad4b758116

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20013 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20438 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21908 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18689 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20244 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23688 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20590 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20185 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20233 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20172 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17398 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22760 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17342 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18194 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24454 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/17408 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18419 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18370 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22443 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/19380 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18961 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16086 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/23417 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18146 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Tests-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4798 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22496 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/24673 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18788 "Built successfully") | | [❌ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5458 "Found 1 jsc stress test failure: microbenchmarks/array-from-object-func.js.lockdown") | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.h:
```diff
@@ -81,7 +82,7 @@ class WebResourceLoadScheduler final : public WebCore::LoaderStrategy {
     void isResourceLoadFinished(WebCore::CachedResource&, CompletionHandler<void(bool)>&&) final;
 
-    class HostInformation {
+    class HostInformation : public CanMakeWeakPtr<HostInformation> {
```
I think we should also make this inherit from CanMakeCheckedPtr and store it CheckedPtr/CheckedRef whenever we're calling functions with it.

cdumez replied:
Ok, fixed.

cdumez committed:
Store fewer raw pointers in containers in Source/WebKitLegacy
https://bugs.webkit.org/show_bug.cgi?id=262188

Reviewed by Ryosuke Niwa.

* Source/WebCore/history/HistoryItem.h:
* Source/WebKitLegacy/Storage/StorageNamespaceImpl.cpp:
(WebKit::localStorageNamespaceMap):
(): Deleted.
* Source/WebKitLegacy/Storage/StorageNamespaceImpl.h:
* Source/WebKitLegacy/Storage/StorageThread.cpp:
(WebCore::activeStorageThreads):
(WebCore::StorageThread::start):
(WebCore::StorageThread::terminate):
(): Deleted.
* Source/WebKitLegacy/Storage/StorageThread.h:
* Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.cpp:
(WebKit::storageNamespaceProviders):
(WebKit::WebStorageNamespaceProvider::WebStorageNamespaceProvider):
(WebKit::WebStorageNamespaceProvider::~WebStorageNamespaceProvider):
(WebKit::WebStorageNamespaceProvider::closeLocalStorage):
(WebKit::WebStorageNamespaceProvider::clearLocalStorageForAllOrigins):
(WebKit::WebStorageNamespaceProvider::clearLocalStorageForOrigin):
(WebKit::WebStorageNamespaceProvider::closeIdleLocalStorageDatabases):
(WebKit::WebStorageNamespaceProvider::syncLocalStorage):
(): Deleted.
* Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.h:
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.cpp:
(WebResourceLoadScheduler::hostForURL):
(WebResourceLoadScheduler::scheduleLoad):
(WebResourceLoadScheduler::remove):
(WebResourceLoadScheduler::crossOriginRedirectReceived):
(WebResourceLoadScheduler::servePendingRequests):
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.h:
* Source/WebKitLegacy/mac/History/BackForwardList.h:
* Source/WebKitLegacy/mac/History/WebBackForwardList.mm:
(kit):
(-[WebBackForwardList initWithBackForwardList:]):
(-[WebBackForwardList dealloc]):
* Source/WebKitLegacy/mac/History/WebHistoryItem.mm:
(-[WebHistoryItem dealloc]):
(-[WebHistoryItem copyWithZone:]):
(core):
(kit):
(-[WebHistoryItem initWithWebCoreHistoryItem:]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebVisitedLinkStore.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebVisitedLinkStore.mm:
(visitedLinkStores):
(WebVisitedLinkStore::WebVisitedLinkStore):
(WebVisitedLinkStore::~WebVisitedLinkStore):
(WebVisitedLinkStore::removeAllVisitedLinks):
* Source/WebKitLegacy/mac/WebView/WebScriptWorld.mm:
(-[WebScriptWorld initWithWorld:]):
(-[WebScriptWorld dealloc]):
(+[WebScriptWorld findOrCreateWorld:]):

Canonical link: https://commits.webkit.org/268556@main
